support/util/expose.go

```diff
@@ -62,3 +62,23 @@ func ServiceExternalDNSHostname(hcp *hyperv1.HostedControlPlane, serviceType hyp
 	}
 	return ""
 }
+
+func ServiceExternalDNSHostnameByHC(hc *hyperv1.HostedCluster, serviceType hyperv1.ServiceType) string {
+	// external DNS hostname can only be set when HC is Public
```

Comment:
This function just mirrors [this ](https://github.com/openshift/hypershift/blob/c001eee2094f6c1b29b238633bf17d98788140a4/support/util/expose.go#L46 
) function but with HC as input, which is used to determine ExternalDNSReachable condition status: 
https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go#L562

I agree, the comment might be misleading, the early return here is to skip ExternalDNS reachability check for private clusters and set the condition to `Unknown`.
Can we actually reach external DNS host in private clusters from the CPO to do the check?

Reply:
Ahh, that makes sense. Yeah, we do create the dns record via external-dns but it points to an IP that we can't reach in the private case.

Please ignore my comment. Thx!

Reply:
shall we then update the comment instead of replicating a misleading one? would
`// external DNS hostname can only be reached when HC is Public` be more accurate?

Reply:
updated the comment